### PR TITLE
Add no-else-return check

### PR DESCRIPTION
**Pull request by 98teg**
Hi! I found your project, and I think it is pretty cool. So I wanted to contribute. I decided to add a rule that seems to be on the TODO list. [This page](https://eslint.org/docs/rules/no-else-return) is what I used as a reference for the expected behaviour. I tried to adapt to the project structure, but please let me know if there is anything that you would like to change. I decided to implement the check as a separate file as it is quite a complex process. Hope you find this useful!

### CODE REVIEW

File: gdtoolkit/linter/__init__.py [hunk 1]
```diff
@@ -81,7 +81,9 @@
         "mixed-tabs-and-spaces": None,
         # misc
         "excluded_directories": {".git"},
-        # no-else-return
+        "no-else-return": {
+            "allow-elif": True
```
Comment (Scony) [inline gdtoolkit/linter/__init__.py:85]: please extract to `no-elif-return` rule in order to:

1. simplify in general (single responsibility principle)
2. allow disabling `no-else-return` without disabling `no-elif-return`

**Review by Scony — CHANGES_REQUESTED** [review]
thanks! just one minor change + CI fixes and this will be ready to go :)

File: gdtoolkit/linter/__init__.py (hunk 1 above)
Comment (98teg) [reply gdtoolkit/linter/__init__.py:85]: I bundle them together as it is how ESLint seems to have done it. But they could be two separate rules too. I first thought that allowing `elif` could influence how to detect an `else` statement to remove. However, writing an example, I realised that it shouldn't. I need to fix that.

I want to keep them both in the same file, as they share logic. I could rename the file from `no_else_return_check.py` to `if_return_checks.py`.

I could also extract the common logic and create two separate files, but the methods are rather specific. What do you prefer?

Comment (Scony) [reply gdtoolkit/linter/__init__.py:85]: `if_return_checks.py` would be good enough :)

**Review by Scony — APPROVED** [review]
thanks!

**Pull request merged**